### PR TITLE
server: Update docker image to bookworm and add CI workflow to test it

### PR DESCRIPTION
**Pull request by anlambert**
Following #1061, update the svix-server docker image to debian bookworm.

Add a CI workflow to test image build and container execution.

### CODE REVIEW

File: .github/workflows/server-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Build server image
+      run: docker-compose build
+      working-directory: ./server
+
+    - name: Run server with docker-compose
+      run: docker-compose up -d
+      working-directory: ./server
+
+    - name: Check server container is up and running
+      run: curl --retry 5 --retry-all-errors http://localhost:8071/api/v1/health/
```
[the remark this answers is not in the document]
Comment (anlambert) [reply .github/workflows/server-docker-image.yml:32]: `curl` uses exponential backoff when retrying so it waits 1 second after the first try, then 2, 4, 8, 16 and finally 32 seconds. It seems sufficient for the server to be up and API responsive.

**Comment by anlambert** [comment]
> use apt instead of apt-get and remove not required package version pins.

It seems the linting job does not like it, will update.

File: .github/workflows/server-docker-image.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tasn) [reply .github/workflows/server-docker-image.yml:32]: Yeah, 100%, great. :)

**Comment by tasn** [comment]
No idea why it's failing again. :)

**Comment by anlambert** [comment]
> No idea why it's failing again. :)

Because I removed the version pins on packages, sigh .. That linting rule is quite weird for debian (only the debian version should be pinned from my point of view) but anyway I will restore them. I removed other workflows on my dev branch when working on updating the docker file so that's why I did not spot these linting issues.

**Comment by tasn** [comment]
I think version pinning is probably a good idea. It removes the mystery from Dockerfiles. E.g. my local testing my have a cached layer with an old zlib, and the release will be with a new zlib that may have issues. So I'm not against that. :)

**Comment by anlambert** [comment]
Package version pins restored, all CI jobs should passed now.

**Comment by tasn** [comment]
Awesome, thanks. Rerunning!

**Comment by tasn** [comment]
New version is out based on the updated dockerfile, thanks again!

**Pull request merged**